core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -126,6 +128,11 @@ public Environment(Settings settings) {
             dataFiles = new Path[]{homeFile.resolve("data")};
             dataWithClusterFiles = new Path[]{homeFile.resolve("data").resolve(ClusterName.clusterNameFromSettings(settings).value())};
         }
+        if (settings.get("path.shared_data") != null) {
+            sharedDataFile = PathUtils.get(cleanPath(settings.get("path.shared_data")));
+        } else {
+            sharedDataFile = homeFile.resolve("data");
```

Comment:
This is the same as `path.data`? Or do you mean it should append the cluster name onto it?